File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/api/groupsv2/CredentialResponse.java
package org.whispersystems.signalservice.api.groupsv2;

import com.fasterxml.jackson.annotation.JsonProperty;

public class CredentialResponse {

  @JsonProperty
  private TemporalCredential[] credentials;

  @JsonProperty
  private TemporalCredential[] callLinkAuthCredentials;

  public TemporalCredential[] getCredentials() {
    return credentials;
  }

  public TemporalCredential[] getCallLinkAuthCredentials() {
    return callLinkAuthCredentials;
  }
}

File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/api/groupsv2/CredentialResponse.kt
/*
 * Copyright 2025 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.whispersystems.signalservice.api.groupsv2

class CredentialResponse(
  val credentials: Array<TemporalCredential> = emptyArray(),
  val callLinkAuthCredentials: Array<TemporalCredential> = emptyArray()
)
